tests/templates/test_layers.py

```diff
@@ -642,3 +644,226 @@ def circuit(weights):
+                    assert isinstance(rec.queue[idx], exp_gate)
+
+                    # check the wires the gates act on
+                    assert rec.queue[idx]._wires == exp_wires[j]
+
+                    # check that parametrized gates takes the parameters \phi and \theta properly
+                    if exp_gate is qml.CRot:
+
+                        if j < idx_CRot:
+                            exp_params = [-phi, np.pi, phi]
+                        if j > idx_CRot:
+                            exp_params = [phi, np.pi, -phi]
+                        if j == idx_CRot:
+                            exp_params = [0, 2 * theta, 0]
+
+                        assert rec.queue[idx].parameters == exp_params
+
+                    elif exp_gate is qml.PhaseShift:
+
+                        if j < idx_CRot:
+                            exp_params = -phi
+                            if j == idx_CRot / 2:
+                                exp_params = phi
+                        if j > idx_CRot:
+                            exp_params = phi
+                            if j == (3 * idx_CRot + 2) / 2:
+                                exp_params = -phi
+
+                        assert rec.queue[idx].parameters == exp_params
+
+    @pytest.mark.parametrize(
+        ("weights", "n_wires", "msg_match"),
+        [
+            (np.ones((4, 2, 2)), 4, "'weights' must be of shape"),
+            (np.ones((4, 3, 1)), 4, "'weights' must be of shape"),
+            (
+                np.ones((4, 3, 1)),
+                1,
+                "This template requires the number of qubits greater than one",
+            ),
+        ],
+    )
+    def test_particle_conserving_u1_exceptions(self, weights, n_wires, msg_match):
+        """Test that ParticleConservingU1 throws an exception if the parameter array have illegal
+        shape."""
+
+        wires = range(n_wires)
+        dev = qml.device("default.qubit", wires=n_wires)
+
+        def circuit(weights=weights, wires=None, init_state=None):
+            ParticleConservingU1(weights=weights, wires=wires, init_state=init_state)
+            return qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(ValueError, match=msg_match):
+            circuit(weights=weights, wires=wires, init_state=np.array([1, 1, 0, 0]))
+
+    def test_integration(self, tol):
+        """Test integration with PennyLane"""
+
+        N = 4
+        wires = range(N)
+        layers = 2
+        weights = np.array(
+            [
+                [[-0.09009989, -0.00090317], [-0.16034551, -0.13278097], [-0.02926428, 0.05175079]],
+                [[-0.07988132, 0.11315495], [-0.16079166, 0.09439518], [-0.04321269, 0.13678911]],
+            ]
+        )
+
+        dev = qml.device("default.qubit", wires=N)
+
+        @qml.qnode(dev)
+        def circuit(weights):
+            ParticleConservingU1(weights, wires, init_state=np.array([1, 1, 0, 0]))
+
+        return [qml.expval(qml.PauliZ(w)) for w in range(N)]
```

Comment:
Note the indentation here: this test now returns `[qml.expval(qml.PauliZ(w)) for w in range(N)]` which causes it to pass by default without evaluating the `assert` statement.
```suggestion
            return [qml.expval(qml.PauliZ(w)) for w in range(N)]
```